### PR TITLE
CI fixes from upstream push

### PR DESCRIPTION
**Pull request by jjsjann123**
1. disables cpp tests on pre-volta device. (regex is easier than macro :stuck_out_tongue:)
2. guard nanosleep for sm < 70

### CODE REVIEW

**Comment by csarofeen** [comment]
Is there any way to bury this in the definition of `TEST(NVFuserTest,`... could we extend `TEST` with `NVFUSER_TEST`?

File: test/cpp/jit/test_gpu.cpp [hunk 1]
```diff
@@ -51,6 +51,12 @@ using namespace at::indexing;
 
 namespace {
 
+inline void skipOnPreVoltaGPU() {
```
Comment (naoyam) [inline test/cpp/jit/test_gpu.cpp]: Missing `GTEST_SKIP`?

**Comment by naoyam** [comment]
Gtest fixtures can be used. @jjsjann123, I created a quick PR (#1306). 

Doc on fixtures:
https://github.com/google/googletest/blob/main/docs/primer.md#test-fixtures-using-the-same-data-configuration-for-multiple-tests-same-data-multiple-tests

**Comment by jjsjann123** [comment]
> Gtest fixtures can be used. @jjsjann123, I created a quick PR (#1306).
> 
> Doc on fixtures: https://github.com/google/googletest/blob/main/docs/primer.md#test-fixtures-using-the-same-data-configuration-for-multiple-tests-same-data-multiple-tests

This looks cleaner. I'm merging yours and fixing it up.

**Comment by jjsjann123** [comment]
Not really cleaned up... lemme do one more push

File: test/cpp/jit/test_gpu.cpp (hunk 1 above)
Comment (naoyam) [reply test/cpp/jit/test_gpu.cpp]: Is this function still used?

File: test/cpp/jit/test_gpu.cpp [hunk 2]
```diff
@@ -103,14 +109,28 @@ bool isPredicated(TensorView* tv, GpuLower& gpulw) {
...
+    skipOnPreVoltaGPU();
+  }
+
+  void skipOnPreVoltaGPU() {
+    if (at::cuda::getDeviceProperties(0)->major < 6) {
+      GTEST_SKIP() << "Volta or newer GPU is required";
+      return;
+    }
+  }
+};
```
Comment (naoyam) [inline test/cpp/jit/test_gpu.cpp]: Why don't we move this to a common header filer for our C++ tests?

Comment (jjsjann123) [reply test/cpp/jit/test_gpu.cpp]: already did this :stuck_out_tongue:

**Comment by jjsjann123** [comment]
Now it should be good for review

**Comment by jjsjann123** [comment]
Note: updated tests passed on volta.

While these fails on PASCAL:
```
[  FAILED  ] 6 tests, listed below:
[  FAILED  ] NVFuserTest.FusionMagicSchedulerBatchNormalization_CUDA
[  FAILED  ] NVFuserTest.FusionSmemPredicateUnswitch_CUDA
[  FAILED  ] NVFuserTest.FusionIssue1133_CUDA
[  FAILED  ] NVFuserTest.FusionIssue1223_CUDA
[  FAILED  ] NVFuserTest.FusionMaxPoolingStrided_CUDA
[  FAILED  ] NVFuserTest.FusionConv2DStaticStrided_CUDA
```

Everything looks like sync issue, which are expected. (I'll turn them off in the old PR #1295.

One thing that will investigate separately are vvv
`FusionMagicSchedulerBatchNormalization_CUDA` seems to be failing on 
```
[ RUN      ] NVFuserTest.FusionMagicSchedulerBatchNormalization_CUDA
unknown file: Failure
C++ exception with description "CUDA driver error: too many resources requested for launch
```

File: test/cpp/jit/test_gpu_shift.cpp [hunk 3]
```diff
@@ -46,6 +47,15 @@ using namespace at::indexing;
 
 namespace {
 
+inline void deviceMajorMinorCheck(int major, int minor = 0) {
```
Comment (naoyam) [inline test/cpp/jit/test_gpu_shift.cpp]: This can be removed

**Review by naoyam — APPROVED** [review]
LGTM

**Pull request merged**